### ISSUE TITLE
F821 false positive for `__class__` in lambda expression in class definition

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#20562]
### Summary

[`undefined-name` (F821)](https://docs.astral.sh/ruff/rules/undefined-name/) has a false positive for `__class__` in a lambda expression within a class definition. [Example](https://play.ruff.rs/abe91404-af96-4ccf-aebb-bd92daa872d4):
```console
$ cat >f821.py <<'# EOF'
class C:
    f = lambda self: __class__
print(C().f().__name__)
# EOF

$ python f821.py
C

$ ruff --isolated check f821.py --select F821 --output-format concise -q
f821.py:2:22: F821 Undefined name `__class__`
```

### Version

ruff 0.13.1 (706be0a6e 2025-09-18)

**Comment by ntBre** [comment]
Ah, thank you! I'm guessing we need to do something like we did for functions in #20024 
https://github.com/astral-sh/ruff/blob/5996dc3f4fd827cb26a17ac5c27481b0ce3277fe/crates/ruff_linter/src/checkers/ast/mod.rs#L1124-L1140
